File: src/1.SRP/reporting/ReportFormatter.ts
import FormatType from './FormatType';

export default class ReportFormatter {
    private formattedOutput: string

    constructor(object: Object, formattype: FormatType) {
        if(formattype === FormatType.XML) {
            this.formattedOutput = this.convertObjectToXML(object);
        } else if (formattype === FormatType.CSV) {
            this.formattedOutput = this.convertObjectToCSV(object);
        }
    }

    protected convertObjectToXML(object: object) : string {
        return "XML: " +object.toString();
    }
    
    protected convertObjectToCSV(object: object) : string {
        return "CSV: " +object.toString();
    }
    
    protected getFormattedValue() : string {
        return this.formattedOutput;
    }
}

File: 1.SRP/src/reporting/ReportFormatter.java
package reporting;

import reporting.FormatType;

public class ReportFormatter {

    private String formattedVal;

    ReportFormatter(Object obj, FormatType formatType) {
        switch (formatType) {
            case CSV: formattedVal = convertObjectToCSV(obj);
                        break;
            case XML: formattedVal = convertObjectToXML(obj);
                        break;
        }
    }
    private String convertObjectToXML(Object object) {
        return "XML + " + object.toString();
    }
    private String convertObjectToCSV(Object object) {
        System.out.println("Convert to CSV");
        return "CSV : " + object.toString();
    }

    public String getFormattedValue() {
        return formattedVal;
    }
}
